xblock/query.py

```diff
@@ -0,0 +1,111 @@
+"""
+    Define Query and Shared objects. Query supports ad-hoc queries via get, find, set operators.
+    Shared uses `bind` functions to access data from pre-defined entities.  
+"""
+
+from copy import deepcopy
+
+class Query(object):
+    """
+    Class for handling remote query operations
+    """
+
+    def __init__(self, field_name, remote_scope, bind):
+        self._queryable = Queryable()
+        self._queryable.bind(field_name, remote_scope, bind)
+
+    def __get__(self, field, field_class):
+        """Get the queryable instance
+        
+        Args:
+            xblock (TYPE): Description
+            xblock_class (TYPE): Description
+        
+        Returns:
+            TYPE: Description
+        """
+
+        return self._queryable
+
+    def __set__(self, field, val):
+        print val
```

Comment:
When there is dummy code like this, it is good to have a TODO comment, so you see a pylint warning/error if you try to commit it.